### ISSUE TITLE
Amazon Linux 2 installation guide incomplete (missing container-selinux)

### ISSUE THREAD

**Issue by pangorgo** [containers/podman.io_old#207]
It seems provided installation steps for Amazon Linux 2 are incomplete.

After adding kubic repo and running `yum install podman` it throws message about missing `container-selinux` dependency:

```
Loaded plugins: extras_suggestions, langpacks, priorities, update-motd
devel_kubic_libcontainers_stable                                                                                                                                                     | 1.3 kB  00:00:00
devel_kubic_libcontainers_stable/primary                                                                                                                                             | 9.8 kB  00:00:00
devel_kubic_libcontainers_stable                                                                                                                                                                      27/27
2 packages excluded due to repository priority protections
Resolving Dependencies
--> Running transaction check
---> Package podman.x86_64 0:1.8.1-2.1.el7 will be installed
--> Processing Dependency: podman-plugins = 1.8.1-2.1.el7 for package: podman-1.8.1-2.1.el7.x86_64
--> Processing Dependency: slirp4netns >= 0.3.0-2 for package: podman-1.8.1-2.1.el7.x86_64
--> Processing Dependency: containernetworking-plugins >= 0.7.5-1 for package: podman-1.8.1-2.1.el7.x86_64
--> Processing Dependency: runc for package: podman-1.8.1-2.1.el7.x86_64
--> Processing Dependency: nftables for package: podman-1.8.1-2.1.el7.x86_64
--> Processing Dependency: containers-common for package: podman-1.8.1-2.1.el7.x86_64
--> Processing Dependency: container-selinux for package: podman-1.8.1-2.1.el7.x86_64
--> Processing Dependency: conmon for package: podman-1.8.1-2.1.el7.x86_64
--> Running transaction check
---> Package conmon.x86_64 2:2.0.11-1.1.el7 will be installed
---> Package containernetworking-plugins.x86_64 0:0.8.5-11.1.el7 will be installed
---> Package containers-common.x86_64 2:0.1.41-7.1.el7 will be installed
---> Package nftables.x86_64 1:0.9.0-14.amzn2.0.1 will be installed
--> Processing Dependency: libnftnl.so.11(LIBNFTNL_11)(64bit) for package: 1:nftables-0.9.0-14.amzn2.0.1.x86_64
--> Processing Dependency: libnftnl.so.11()(64bit) for package: 1:nftables-0.9.0-14.amzn2.0.1.x86_64
---> Package podman.x86_64 0:1.8.1-2.1.el7 will be installed
--> Processing Dependency: container-selinux for package: podman-1.8.1-2.1.el7.x86_64
---> Package podman-plugins.x86_64 0:1.8.1-2.1.el7 will be installed
---> Package runc.x86_64 0:1.0.0-0.1.20190510.git2b18fe1.amzn2 will be installed
---> Package slirp4netns.x86_64 0:0.4.3-22.1.el7 will be installed
--> Running transaction check
---> Package libnftnl.x86_64 0:1.1.1-4.amzn2 will be installed
---> Package podman.x86_64 0:1.8.1-2.1.el7 will be installed
--> Processing Dependency: container-selinux for package: podman-1.8.1-2.1.el7.x86_64
--> Finished Dependency Resolution
Error: Package: podman-1.8.1-2.1.el7.x86_64 (devel_kubic_libcontainers_stable)
           Requires: container-selinux
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

**Comment by TomSweeneyRedHat** [comment]
@lsm5 do you know what needs adjusting?

**Comment by lsm5** [comment]
@pangorgo can you check if container-selinux is available in the rhel extras channel? I'm guessing Amazon linux would also be providing a clone of the rhel extras repo just like CentOS provides CentOS-Extras. If it does, we can update the docs with the steps to fetch it. If not, I'll look at adding container-selinux to OBS.

**Comment by lsm5** [comment]
@pangorgo try this:
```
cd /etc/yum.repos.d/
sudo wget https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable/CentOS_7/devel:kubic:libcontainers:stable.repo
sudo yum -y install yum-plugin-copr
sudo yum -y copr enable lsm5/container-selinux
sudo yum -y install podman
```
worked for me on amazon linux container. Once you verify, I'll PR to update the install doc.

**Comment by ArangoGutierrez** [comment]
I am still getting the same issue as the one reported in this issue
```bash
           Available: docker-19.03.6ce-2.amzn2.x86_64 (amzn2extra-docker)
               Not found
           Available: docker-19.03.6ce-3.amzn2.x86_64 (amzn2extra-docker)
               Not found
           Available: docker-19.03.6ce-4.amzn2.x86_64 (amzn2extra-docker)
               Not found
           Available: docker-19.03.13ce-1.amzn2.x86_64 (amzn2extra-docker)
               Not found
           Available: docker-20.10.4-1.amzn2.x86_64 (amzn2extra-docker)
               Not found
Error: Package: fuse-overlayfs-1.5.0-1.el7.5.1.x86_64 (devel_kubic_libcontainers_stable)
           Requires: libfuse3.so.3(FUSE_3.2)(64bit)
Error: Package: fuse-overlayfs-1.5.0-1.el7.5.1.x86_64 (devel_kubic_libcontainers_stable)
           Requires: fuse3
Error: Package: fuse-overlayfs-1.5.0-1.el7.5.1.x86_64 (devel_kubic_libcontainers_stable)
           Requires: libfuse3.so.3(FUSE_3.0)(64bit)
Error: Package: fuse-overlayfs-1.5.0-1.el7.5.1.x86_64 (devel_kubic_libcontainers_stable)
           Requires: libfuse3.so.3()(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```
when following those recommended lines

**Comment by TomSweeneyRedHat** [comment]
@lsm5 further thoughts?

**Comment by ArangoGutierrez** [comment]
after working with @lsm5 , he was able to get 
```bash
Version:            1.6.4
RemoteAPI Version:  1
Go Version:         go1.12.12
OS/Arch:            linux/amd64
```
following this steps:

1.Have an /etc/yum.repos.d/centos.repo that looks like:
```
[centos-base]
name=CentOS-7 - Base
baseurl=http://mirror.centos.org/centos-7/7/os/x86_64/
enabled=0
gpgcheck=0
[centos-extras]
name=CentOS-7 - Extras
baseurl=http://mirror.centos.org/centos-7/7/extras/x86_64/
gpgcheck=0
```
2. In `/etc/yum.repos.d/amzn2-core.repo`  comment out `priority=10 line for [amzn2-core]` repo
3. `sudo yum -y install podman --enablerepo=centos-base`

**Comment by lsm5** [comment]
just for some more clarity to anyone reading this, the steps in the above comment will help you get the last shipped centos7/rhel7 version of podman, not the latest podman available upstream.

**Comment by KCSesh** [comment]
I am actually seeing an issue on AL2 as well: 

```
--> Finished Dependency Resolution
Error: Package: podman-3.0.1-2.el7.3.1.x86_64 (devel_kubic_libcontainers_stable)
           Requires: runc >= 2:1.0.0-148.rc93
           Available: docker-17.06.2ce-1.102.amzn2.x86_64 (amzn2extra-docker)
               runc = 0-1.git810190c
           Available: docker-17.12.1ce-2.amzn2.x86_64 (amzn2extra-docker)
               runc = 0-1.git9f9c962
           Available: docker-18.03.1ce-2.amzn2.x86_64 (amzn2extra-docker)
               runc = 0-1.git4fc53a8
           Available: docker-18.03.1ce-3.amzn2.x86_64 (amzn2extra-docker)
               runc = 0-1.git4fc53a8
           Available: docker-18.03.1ce-5.amzn2.x86_64 (amzn2extra-docker)
               runc = 0-1.git4fc53a8
           Available: docker-18.06.1ce-2.amzn2.x86_64 (amzn2extra-docker)
               runc = 0-1.git69663f0
           Available: docker-18.06.1ce-4.amzn2.x86_64 (amzn2extra-docker)
               runc = 0-1.git69663f0
           Available: docker-18.06.1ce-5.amzn2.x86_64 (amzn2extra-docker)
               runc = 0-1.git69663f0
           Available: docker-18.06.1ce-6.amzn2.x86_64 (amzn2extra-docker)
               runc = 0-1.git69663f0
           Available: docker-18.06.1ce-7.amzn2.x86_64 (amzn2extra-docker)
               runc = 0-1.git69663f0
           Available: docker-18.06.1ce-8.amzn2.x86_64 (amzn2extra-docker)
               runc = 0-1.git69663f0
           Available: docker-18.06.1ce-10.amzn2.x86_64 (amzn2extra-docker)
               runc = 0-1.git69663f0
           Available: runc-1.0.0-0.1.20190510.git2b18fe1.amzn2.x86_64 (amzn2extra-docker)
               runc = 1.0.0-0.1.20190510.git2b18fe1.amzn2
           Available: runc-1.0.0-0.1.20200204.gitdc9208a.amzn2.x86_64 (amzn2extra-docker)
               runc = 1.0.0-0.1.20200204.gitdc9208a.amzn2
           Available: runc-1.0.0-0.1.20200826.gitff819c7.amzn2.x86_64 (amzn2extra-docker)
               runc = 1.0.0-0.1.20200826.gitff819c7.amzn2
           Available: runc-1.0.0-0.1.20210225.git12644e6.amzn2.x86_64 (amzn2extra-docker)
               runc = 1.0.0-0.1.20210225.git12644e6.amzn2
           Available: docker-18.09.9ce-2.amzn2.x86_64 (amzn2extra-docker)
               Not found
           Available: docker-19.03.6ce-1.amzn2.x86_64 (amzn2extra-docker)
               Not found
           Available: docker-19.03.6ce-2.amzn2.x86_64 (amzn2extra-docker)
               Not found
           Available: docker-19.03.6ce-3.amzn2.x86_64 (amzn2extra-docker)
               Not found
           Available: docker-19.03.6ce-4.amzn2.x86_64 (amzn2extra-docker)
               Not found
           Available: docker-19.03.13ce-1.amzn2.x86_64 (amzn2extra-docker)
               Not found
           Available: docker-20.10.4-1.amzn2.x86_64 (amzn2extra-docker)
               Not found
Error: Package: 4:containers-common-1-17.el7.17.1.noarch (devel_kubic_libcontainers_stable)
           Requires: oci-runtime
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
$ 

```


I followed these steps suggested by @ArangoGutierrez : 

1.Have an /etc/yum.repos.d/centos.repo that looks like:
```
[centos-base]
name=CentOS-7 - Base
baseurl=http://mirror.centos.org/centos-7/7/os/x86_64/
enabled=0
gpgcheck=0
[centos-extras]
name=CentOS-7 - Extras
baseurl=http://mirror.centos.org/centos-7/7/extras/x86_64/
gpgcheck=0
```

In `/etc/yum.repos.d/amzn2-core.repo` comment out `priority=10 line for [amzn2-core]` repo

`sudo yum -y install podman --enablerepo=centos-base`


---

Runc did not exist on my system.

I tried installing the latest runc manually based on the error, 
but I still am seeing the same error. 
https://github.com/opencontainers/runc/releases/tag/v1.0.0-rc94

```
$ which runc
/usr/bin/runc
$ runc --version
runc version 1.0.0-rc94+dev
commit: 4e50d1c16334d42506f1aa8bfadbdabf785e8acf
spec: 1.0.2-dev
go: go1.15.8
libseccomp: 2.4.1

```

**Comment by rhatdan** [comment]
Complain to Docker Inc to use the systems runc rather then requiring their own, or if they require their own, install it in a not system default directory, so the distributions runc can be used.

**Comment by lsm5** [comment]
@KCSesh the steps that @ArangoGutierrez mentioned will only work on AL2 for the last shipped podman in CentOS Extras (*not* Kubic). The `devel_kubic_libcontainers_stable` won't help on AL2 for podman. Consider building from source or switching to a newer distro.

**Comment by KCSesh** [comment]
@lsm5 You mind giving more details? 

I just spun up an AL2 instance to try and install podman. 

Are you saying that I need to configure the instance to use the CentOS Extras repos and not Kubic? 
I can modify the configuration of the repos if needed. 

I guess I am wondering what the difference is between my AL2 instance and @ArangoGutierrez AL2 instance.

**Comment by lsm5** [comment]
> @lsm5 You mind giving more details?

See: https://podman.io/blogs/2021/03/02/podman-support-for-older-distros.html . Whatever's said in there for CentOS 7 also applies to AL2.

> 
> I just spun up an AL2 instance to try and install podman.
> 
> Are you saying that I need to configure the instance to use the CentOS Extras repos and not Kubic?
> I can modify the configuration of the repos if needed.
> 
> I guess I am wondering what the difference is between my AL2 instance and @ArangoGutierrez AL2 instance.


The steps that @ArangoGutierrez mentioned will only be good enough for getting you the podman version that was last shipped on CentOS7. In other words, you won't get anything newer than v1.6.2.

If you want anything newer than v1.6.2 on AL2, your best (if not, only) option would be to build podman from upstream source.

In your logs above, I noticed you had repos from the Kubic project added. The podman from those repos that was last shipped for CentOS 7 /AL2 won't be installable anymore as you experienced above as the dependencies have moved ahead and the steps are no longer guaranteed to work.

Long story short, on AL2, there's no point using Kubic project repos for podman.

HTH.

**Comment by KCSesh** [comment]
Thanks @lsm5 that was helpful.

I was able to successfully build podman from source on AL2. 

```
podman --version
podman version 3.2.0-dev
```
